docs/advanced-features/middleware.md

```diff
@@ -1,95 +1,166 @@
+7. Dynamic Routes (`/blog/[slug]`)
+8. `fallback` (`rewrites`) from `next.config.js`
 
-These native Web API objects are extended to give you more control over how you manipulate and configure a response, based on the incoming requests.
+There are two ways to define which paths Middleware will run on: 
 
-The function signature:
+1. Custom matcher config
+2. Conditional statements
 
-```ts
-import type { NextFetchEvent } from 'next/server'
-import type { NextRequest } from 'next/server'
+### Matcher
+
+`matcher` allows you to filter Middleware to run on specific paths. 
 
-export type Middleware = (
-  request: NextRequest,
-  event: NextFetchEvent
-) => Promise<Response | undefined> | Response | undefined
+```js
+export const config = {
+  matcher: '/about/:path*',
+}
 ```
 
-The function can be a default export and as such, does **not** have to be named `middleware`. Though this is a convention. Also note that you only need to make the function `async` if you are running asynchronous code.
+You can match a single path or multiple paths with an array syntax:
 
-Read the full [Middleware API reference](/docs/api-reference/edge-runtime.md), note [Node.js APIs are not supported in this environment](/docs/api-reference/edge-runtime.md#unsupported-apis)
+```js
+export const config = {
+  matcher: ['/about/:path*', '/dashboard/:path*'],
+}
+```
 
-## Examples
+### Conditional Statements
 
-Middleware can be used for anything that shares logic for a set of pages, including:
+```typescript
+// middleware.ts
 
-- [Authentication](https://github.com/vercel/examples/tree/main/edge-functions)
-- [Bot protection](https://github.com/vercel/examples/tree/main/edge-functions)
-- [Redirects and rewrites](https://github.com/vercel/examples/tree/main/edge-functions)
-- [Handling unsupported browsers](https://github.com/vercel/examples/tree/main/edge-functions/user-agent-based-rendering)
-- [Feature flags and A/B tests](https://github.com/vercel/examples/tree/main/edge-functions)
-- [Advanced i18n routing requirements](https://github.com/vercel/examples/tree/main/edge-functions)
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
 
-## Execution Order
+export function middleware(request: NextRequest) {
+  if (request.nextUrl.pathname.startsWith('/about')) {
+    return NextResponse.rewrite(new URL('/about-2', request.url))
+  }
 
-Middleware runs directly after `redirects` and `headers`, before the first filesystem lookup. This excludes `/_next` files.
+  if (request.nextUrl.pathname.startsWith('/dashboard')) {
+    return NextResponse.rewrite(new URL('/dashboard/user', request.url))
+  }
+}
+```
 
-## Deployment
+## NextResponse
 
-Middleware uses a [strict runtime](/docs/api-reference/edge-runtime.md) that supports standard Web APIs like `fetch`. This works out of the box using `next start`, as well as on Edge platforms like Vercel, which use [Edge Functions](http://www.vercel.com/edge).
```

Comment:
What is happening with edge functions? Is the relationship between middleware and edge functions changing? We should also address this since “edge functions” is a fairly popular term, especially since we’ve been using it interchangeably with middleware for a while

Is this repo still updated and relevant? If so, we might want to link to it: https://github.com/vercel/examples/tree/main/edge-functions

Reply:
This is more of a Vercel thing than a Next.js thing 😄 Edge Functions and Edge Middleware are two separate things (previously poorly conflated).